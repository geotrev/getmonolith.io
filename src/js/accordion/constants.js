export const Selectors = {
  // unique
  DATA_ACCORDION: "data-accordion",
  DATA_COLLAPSIBLE: "data-collapsible",
  DATA_TOGGLE_MULTIPLE: "data-toggle-multiple",
  // common
  DATA_PARENT: "data-parent",
  DATA_VISIBLE: "data-visible",
  DATA_TARGET: "data-target",
  // accessibility
  ARIA_EXPANDED: "aria-expanded",
  ARIA_HIDDEN: "aria-hidden",
}

export const Messages = {
  NO_ACCORDION_ID_ERROR:
    "Could not initialize accordion; you must include a value for the 'data-accordion' attribute.",
  NO_ACCORDION_ERROR: id => `Could not find element matching [data-accordion='${id}']`,
  TRIGGERS_TO_COLLAPSIBLES_LENGTH_WARNING: (id, triggersLength, collapsiblesLength) =>
    `Your accordion with id '${id}' has ${triggersLength} triggers and ${collapsiblesLength} collapsibles; make sure there is a trigger for each collapsible!`,
}
