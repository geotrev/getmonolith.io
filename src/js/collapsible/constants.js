export const Selectors = {
  // unique
  DATA_COLLAPSIBLE: "data-collapsible",
  // common
  DATA_VISIBLE: "data-visible",
  DATA_TARGET: "data-target",
  DATA_PARENT: "data-parent",
  // accessibility
  ARIA_EXPANDED: "aria-expanded",
  ARIA_CONTROLS: "aria-controls",
  ARIA_HIDDEN: "aria-hidden",
  ARIA_LABELLEDBY: "aria-labelledby",
  TABINDEX: "tabindex",
  // classes
  TRIGGER_READY_CLASS: "collapsible-trigger-ready",
  CONTENT_READY_CLASS: "collapsible-content-ready",
}

export const CssProperties = {
  HEIGHT: "height",
  VISIBILITY: "visibility",
}

export const CssValues = {
  AUTO: "auto",
  HIDDEN: "hidden",
  VISIBLE: "visible",
}

export const Events = {
  CLICK: "click",
  TRANSITIONEND: "transitionend",
}

export const Messages = {
  NO_COLLAPSIBLE_ID_ERROR:
    "Could not initialize collapsible; you must include a value for the 'data-collapsible' attribute.",
  NO_TRIGGER_ERROR: id =>
    `Could not find collapsible trigger with [data-target='${id}']; you can't have a collapsible without a trigger.`,
  NO_TRIGGER_ID_ERROR: id => `Could not find id on collapsible trigger with [data-target='${id}'].`,
  NO_CONTENT_ERROR: id =>
    `Could not find collapsible content with id '${id}'; you can't have a collapsible without content.`,
}
