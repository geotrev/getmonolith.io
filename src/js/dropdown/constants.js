export const KeyCodes = {
  TAB: 9,
  SHIFT: 16,
  ESCAPE: 27,
  ARROW_UP: 38,
  ARROW_DOWN: 40,
}

export const Selectors = {
  // unique
  DATA_DROPDOWN: "data-dropdown",
  DROPDOWN_MENU_CLASS: "dropdown-menu",
  // common
  DATA_TARGET: "data-target",
  DATA_PARENT: "data-parent",
  DATA_VISIBLE: "data-visible",
  // accessibility
  TABINDEX: "tabindex",
  ARIA_HASPOPUP: "aria-haspopup",
  ARIA_CONTROLS: "aria-controls",
  ARIA_LABELLEDBY: "aria-labelledby",
  ARIA_EXPANDED: "aria-expanded",
  // classes
  OVERLAY_OPEN: "overlay-open",
}

export const Events = {
  KEYDOWN: "keydown",
  CLICK: "click",
  BLUR: "blur",
}

export const Messages = {
  NO_DROPDOWN_ID_ERROR:
    "Could not setup dropdown. Make sure it has a valid [data-dropdown] attribute with a unique id as its value.",
  NO_MENU_ERROR: attr => `Could not find menu associated with ${attr}.`,
  NO_DROPDOWN_ITEMS_ERROR: attr => `Could not find any list items associated with ${attr}.`,
  NO_DROPDOWN_ACTIONS_ERROR: attr =>
    `Could not find any button or anchor elements associated with ${attr}.`,
  NO_PARENT_ERROR: "Could not find dropdown button's [data-parent] attribute.",
}
