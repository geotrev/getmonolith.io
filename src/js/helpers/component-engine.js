import { queryAll } from "./dom"
import { isBrowserEnv } from "./is-browser-env"

const isString = value => typeof value === "string"
const isFunction = value => typeof value === "function"

export const ComponentEngine = {
  /**
   * Initialize an Undernet component globally or by id.
   * @param {Object} metadata
   * @property {string} id
   * @property {string} attribute
   * @property {Object} thisArg
   */
  start: (metadata = {}) => {
    if (!isBrowserEnv) return

    const { id, attribute, thisArg } = metadata

    if (id && isString(id)) {
      const instance = document.querySelector(`[${attribute}='${id}']`)
      if (!instance) return

      const validComponent = [instance].filter(thisArg._validate)[0]
      if (!validComponent) return

      thisArg._components.push(validComponent)
    } else if (!id && !thisArg._components.length) {
      const instances = queryAll(`[${attribute}]`)
      if (!instances.length) return

      const validComponents = instances.filter(thisArg._validate)
      thisArg._components = thisArg._components.concat(validComponents)
    } else {
      // attempted to .start() when .stop() wasn't run,
      // OR tried to instantiate a component that's already active.
    }
  },
  /**
   * Teardown an Undernet component globally or by id.
   * @param {Object} metadata
   * @property {string} id
   * @property {string} attribute
   * @property {Object} thisArg
   * @property {string} activeNodeKey
   * @property {*=} cancelActiveFn
   */
  stop: (metadata = {}) => {
    if (!isBrowserEnv) return

    const { id, attribute, thisArg, activeNodeKey, cancelActiveFn } = metadata

    if (id && isString(id)) {
      let targetIndex
      const instance = thisArg._components.filter((activeInstance, index) => {
        if (activeInstance.getAttribute(attribute) !== id) return false
        targetIndex = index
        return true
      })[0]

      if (!instance) return

      if (
        thisArg[activeNodeKey] &&
        instance === thisArg[activeNodeKey] &&
        isFunction(cancelActiveFn)
      )
        thisArg[cancelActiveFn]()

      thisArg._teardown(instance)
      thisArg._components.splice(targetIndex, 1)
    } else if (!id && thisArg._components.length) {
      if (thisArg[activeNodeKey] && isFunction(cancelActiveFn)) thisArg[cancelActiveFn]()

      thisArg._components.forEach(thisArg._teardown)
      thisArg._components = []
    }
  },
}
