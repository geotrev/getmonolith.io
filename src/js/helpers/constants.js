export const KeyCodes = {
  SHIFT: 16,
  TAB: 9,
  ARROW_UP: 38,
  ARROW_DOWN: 40,
}

export const Selectors = {
  FOCUSABLE_TAGS: ["a", "button", "input", "object", "select", "textarea", "[tabindex]"],
  KEYBOARD_CLASS: "using-keyboard",
  NOT_VISUALLY_HIDDEN_CLASS: ":not(.is-visually-hidden)",
  TABINDEX: "tabindex",
}

export const Events = {
  KEYDOWN: "keydown",
  CLICK: "click",
  BLUR: "blur",
}

export const Messages = {
  NO_SELECTOR_STRING_OR_CHILDREN_ERROR:
    "createFocusTrap must be given one or both of: first parameter (as selector string)" +
    " and/or options.children (array of elements).",
  OPTION_MATCHERS_DATA_TYPE_ERROR:
    "Invalid data type given to options.matchers for createFocusTrap. Expected: Array.",
  INCORRECT_MATCHER_TYPE_ERROR: type =>
    `Invalid matcher given to options.matchers for createFocusTrap. Expected: String. Recieved: ${type}.`,
  NO_MATCHER_LENGTH_ERROR:
    "Invalid value given to options.matchers for createFocusTrap; value must be an array with at least one selector string",
}
