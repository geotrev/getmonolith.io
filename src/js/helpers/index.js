export * from "./component-engine"
export { createFocusRing } from "./create-focus-ring"
export { createFocusTrap } from "./create-focus-trap"
export * from "./dom"
export { focusOnce } from "./focus-once"
export { isBrowserEnv } from "./is-browser-env"
export { isiOSMobile } from "./is-ios-mobile"
export { log } from "./log"
export { throttle } from "./throttle"
