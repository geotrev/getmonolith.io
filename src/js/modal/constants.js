export const KeyCodes = {
  ESCAPE: 27,
}

export const Selectors = {
  // unique
  DATA_MODAL: "data-modal",
  // common
  DATA_TARGET: "data-target",
  DATA_VISIBLE: "data-visible",
  DATA_CLOSE: "data-close",
  DATA_PARENT: "data-parent",
  // accessibility
  ARIA_HIDDEN: "aria-hidden",
  ARIA_MODAL: "aria-modal",
  ROLE: "role",
  TABINDEX: "tabindex",
  // classes
  NO_SCROLL: "no-scroll",
}

export const CssProperties = {
  PADDING_RIGHT: "paddingRight",
  PADDING_LEFT: "paddingLeft",
  CURSOR: "cursor",
}

export const CssValues = {
  AUTO: "auto",
  POINTER: "pointer",
}

export const Events = {
  KEYDOWN: "keydown",
  CLICK: "click",
  RESIZE: "resize",
  BLUR: "blur",
}

export const Messages = {
  NO_TRIGGER_ERROR: id => `Could not find modal trigger with id ${id}.`,
  NO_ID_ERROR:
    "Could not detect an id on your [data-modal] element. Please add a value matching the modal trigger's [data-parent] attribute.",
  NO_MODAL_DIALOG_ERROR: id => `Could not find element with attribute [data-parent='${id}'].`,
}
