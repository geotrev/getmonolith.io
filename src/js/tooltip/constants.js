export const KeyCodes = {
  ESCAPE: 27,
}

export const Selectors = {
  // unique
  DATA_TOOLTIP: "data-tooltip",
  // common
  DATA_VISIBLE: "data-visible",
  DATA_TARGET: "data-target",
  // accessibility
  ROLE: "role",
  ARIA_DESCRIBEDBY: "aria-describedby",
  // classes
  DROP_INLINE_START_CLASS: "is-drop-inline-start",
  DROP_INLINE_END_CLASS: "is-drop-inline-end",
  OVERLAY_OPEN: "overlay-open",
}

export const CssProperties = {
  HEIGHT: "height",
  WIDTH: "width",
  TOP: "top",
  LEFT: "left",
}

export const Events = {
  CLICK: "click",
  MOUSEOVER: "mouseover",
  MOUSEOUT: "mouseout",
  FOCUS: "focus",
  BLUR: "blur",
  KEYDOWN: "keydown",
}

export const Messages = {
  NO_ID_ERROR: "Could not find tooltip id.",
  NO_TRIGGER_ERROR: id => `Could not find a tooltip trigger with attribute [data-target='${id}'].`,
  NO_TOOLTIP_ERROR: id => `Could not find a tooltip with id '${id}'.`,
}
