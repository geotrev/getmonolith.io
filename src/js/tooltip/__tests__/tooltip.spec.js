import Undernet from "../../"
import { find, renderDOM, simulateMouseEvent, simulateKeyboardEvent } from "../../helpers/test"
import { KeyCodes, Messages, Selectors } from "../constants"

const dom = `
  <span class="tooltip" data-tooltip="tooltip-id-1">
    <button class="tooltip-trigger" data-target="tooltip-id-1">Tooltip Button</button>
    <div class="tooltip-box" id="tooltip-id-1">
      This is a tooltip.
    </div>
  </span>
  <span class="tooltip" data-tooltip="tooltip-id-2">
    <button class="tooltip-trigger" data-target="tooltip-id-2">Tooltip Button</button>
    <div class="tooltip-box" id="tooltip-id-2">
      This is a tooltip.
    </div>
  </span>
`

console.error = jest.fn()

describe("Tooltip", () => {
  afterEach(() => {
    Undernet.Tooltips.stop()
  })

  describe("API start", () => {
    it("sets attributes", () => {
      // Given
      const wrapper = renderDOM(dom)
      // When
      Undernet.Tooltips.start()
      // Then
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe("API stop + Focus/Mouseover", () => {
    let wrapper
    let trigger

    beforeEach(() => {
      wrapper = renderDOM(dom)
      trigger = find("[data-target='tooltip-id-1']")

      Undernet.Tooltips.start()
      Undernet.Tooltips.stop()
    })

    it("does not open tooltip on focus", () => {
      trigger.focus()
      expect(wrapper).toMatchSnapshot()
    })

    it("does not open tooltip on mouseover", () => {
      simulateMouseEvent("mouseover", trigger)
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe("#handleEvent", () => {
    let wrapper
    let trigger

    beforeEach(() => {
      wrapper = renderDOM(dom)
      trigger = find("[data-target='tooltip-id-1']")

      Undernet.Tooltips.start()
    })

    it("opens tooltip on focus", () => {
      trigger.focus()
      expect(wrapper).toMatchSnapshot()
    })

    it("opens tooltip on mouseover", () => {
      simulateMouseEvent("mouseover", trigger)
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe("#handleClose (Blur/Mouseout)", () => {
    let wrapper
    let trigger1
    let trigger2

    beforeEach(() => {
      wrapper = renderDOM(dom)
      trigger1 = find("[data-target='tooltip-id-1']")
      trigger2 = find("[data-target='tooltip-id-2']")

      Undernet.Tooltips.start()
    })

    it("hides tooltip on mouseout event", () => {
      // When
      simulateMouseEvent("mouseover", trigger1)
      simulateMouseEvent("mouseout", trigger1)
      // Then
      expect(wrapper).toMatchSnapshot()
    })

    it("hides first tooltip if second is focused", () => {
      // When
      trigger1.focus()
      trigger2.focus()
      // Then
      expect(wrapper).toMatchSnapshot()
    })

    it("will hide hovered tooltip when second is focused", () => {
      // When
      simulateMouseEvent("mouseover", trigger1)
      trigger2.focus()
      // Then
      expect(wrapper).toMatchSnapshot()
    })

    it("will hide focused tooltip when second is hovered", () => {
      // When
      trigger1.focus()
      simulateMouseEvent("mouseover", trigger2)
      // Then
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe("#handleEscapeKeyPress", () => {
    it("closes tooltip", () => {
      // Given
      const wrapper = renderDOM(dom)
      const trigger = find("[data-target='tooltip-id-1']")
      // When
      Undernet.Tooltips.start()
      trigger.focus()
      simulateKeyboardEvent(KeyCodes.ESCAPE)
      // Then
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe("Drop Direction", () => {
    let wrapper
    let trigger1
    let tooltip

    beforeEach(() => {
      wrapper = renderDOM(dom)
      trigger1 = find("[data-target='tooltip-id-1']")
      tooltip = find("#tooltip-id-1")

      Undernet.Tooltips.start()
    })

    it("renders left", () => {
      // Given
      tooltip.classList.add(Selectors.DROP_INLINE_START_CLASS)
      // When
      trigger1.focus()
      // Then
      expect(wrapper).toMatchSnapshot()
    })

    it("renders right", () => {
      // Given
      tooltip.classList.add(Selectors.DROP_INLINE_END_CLASS)
      // When
      trigger1.focus()
      // Then
      expect(wrapper).toMatchSnapshot()
    })

    it("renders bottom", () => {
      // Given
      tooltip.classList.add("is-drop-block-end")
      // When
      trigger1.focus()
      // Then
      expect(wrapper).toMatchSnapshot()
    })
  })
})

const errorDom = (target, tooltip, id) => `
  <span class="tooltip" data-tooltip="${tooltip}">
    <button class="tooltip-trigger" data-target="${target}">Tooltip Button</button>
    <div class="tooltip-box" id="${id}">
      This is a tooltip.
    </div>
  </span>
`

describe("Tooltip Warnings", () => {
  const TOOLTIP_ID = "tooltip-id"

  it("prints console error if [data-tooltip] is empty", () => {
    // Given
    renderDOM(errorDom(TOOLTIP_ID, "", TOOLTIP_ID))
    // When
    Undernet.Tooltips.start()
    // Then
    expect(console.error).toBeCalledWith(Messages.NO_ID_ERROR)
  })

  it("prints console error if [data-target] attribute does not match its parent [data-tooltip]", () => {
    // Given
    renderDOM(errorDom("", TOOLTIP_ID, TOOLTIP_ID))
    // When
    Undernet.Tooltips.start()
    // Then
    expect(console.error).toBeCalledWith(Messages.NO_TRIGGER_ERROR(TOOLTIP_ID))
  })

  it("prints console error if tooltip container's [id] does not match its parent [data-tooltip]", () => {
    // Given
    renderDOM(errorDom(TOOLTIP_ID, TOOLTIP_ID, ""))
    // When
    Undernet.Tooltips.start()
    // Then
    expect(console.error).toBeCalledWith(Messages.NO_TOOLTIP_ERROR(TOOLTIP_ID))
  })
})
