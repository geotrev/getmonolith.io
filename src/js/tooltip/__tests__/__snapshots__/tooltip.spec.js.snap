// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip #handleClose (Blur/Mouseout) hides first tooltip if second is focused 1`] = `
<body>
  
  
  <span
    class="tooltip"
    data-tooltip="tooltip-id-1"
  >
    
    
    <button
      aria-describedby="tooltip-id-1"
      class="tooltip-trigger"
      data-target="tooltip-id-1"
    >
      Tooltip Button
    </button>
    
    
    <div
      class="tooltip-box"
      data-visible="false"
      id="tooltip-id-1"
      role="tooltip"
      style="left: 0px;"
    >
      
      This is a tooltip.
    
    </div>
    
  
  </span>
  
  
  <span
    class="tooltip"
    data-tooltip="tooltip-id-2"
  >
    
    
    <button
      aria-describedby="tooltip-id-2"
      class="tooltip-trigger"
      data-target="tooltip-id-2"
    >
      Tooltip Button
    </button>
    
    
    <div
      class="tooltip-box"
      data-visible="true"
      id="tooltip-id-2"
      role="tooltip"
      style="left: 0px;"
    >
      
      This is a tooltip.
    
    </div>
    
  
  </span>
  

</body>
`;

exports[`Tooltip #handleClose (Blur/Mouseout) hides tooltip on mouseout event 1`] = `
<body>
  
  
  <span
    class="tooltip"
    data-tooltip="tooltip-id-1"
  >
    
    
    <button
      aria-describedby="tooltip-id-1"
      class="tooltip-trigger"
      data-target="tooltip-id-1"
    >
      Tooltip Button
    </button>
    
    
    <div
      class="tooltip-box"
      data-visible="false"
      id="tooltip-id-1"
      role="tooltip"
      style="left: 0px;"
    >
      
      This is a tooltip.
    
    </div>
    
  
  </span>
  
  
  <span
    class="tooltip"
    data-tooltip="tooltip-id-2"
  >
    
    
    <button
      aria-describedby="tooltip-id-2"
      class="tooltip-trigger"
      data-target="tooltip-id-2"
    >
      Tooltip Button
    </button>
    
    
    <div
      class="tooltip-box"
      id="tooltip-id-2"
      role="tooltip"
    >
      
      This is a tooltip.
    
    </div>
    
  
  </span>
  

</body>
`;

exports[`Tooltip #handleClose (Blur/Mouseout) will hide focused tooltip when second is hovered 1`] = `
<body>
  
  
  <span
    class="tooltip"
    data-tooltip="tooltip-id-1"
  >
    
    
    <button
      aria-describedby="tooltip-id-1"
      class="tooltip-trigger"
      data-target="tooltip-id-1"
    >
      Tooltip Button
    </button>
    
    
    <div
      class="tooltip-box"
      data-visible="false"
      id="tooltip-id-1"
      role="tooltip"
      style="left: 0px;"
    >
      
      This is a tooltip.
    
    </div>
    
  
  </span>
  
  
  <span
    class="tooltip"
    data-tooltip="tooltip-id-2"
  >
    
    
    <button
      aria-describedby="tooltip-id-2"
      class="tooltip-trigger"
      data-target="tooltip-id-2"
    >
      Tooltip Button
    </button>
    
    
    <div
      class="tooltip-box"
      data-visible="true"
      id="tooltip-id-2"
      role="tooltip"
      style="left: 0px;"
    >
      
      This is a tooltip.
    
    </div>
    
  
  </span>
  

</body>
`;

exports[`Tooltip #handleClose (Blur/Mouseout) will hide hovered tooltip when second is focused 1`] = `
<body>
  
  
  <span
    class="tooltip"
    data-tooltip="tooltip-id-1"
  >
    
    
    <button
      aria-describedby="tooltip-id-1"
      class="tooltip-trigger"
      data-target="tooltip-id-1"
    >
      Tooltip Button
    </button>
    
    
    <div
      class="tooltip-box"
      data-visible="false"
      id="tooltip-id-1"
      role="tooltip"
      style="left: 0px;"
    >
      
      This is a tooltip.
    
    </div>
    
  
  </span>
  
  
  <span
    class="tooltip"
    data-tooltip="tooltip-id-2"
  >
    
    
    <button
      aria-describedby="tooltip-id-2"
      class="tooltip-trigger"
      data-target="tooltip-id-2"
    >
      Tooltip Button
    </button>
    
    
    <div
      class="tooltip-box"
      data-visible="true"
      id="tooltip-id-2"
      role="tooltip"
      style="left: 0px;"
    >
      
      This is a tooltip.
    
    </div>
    
  
  </span>
  

</body>
`;

exports[`Tooltip #handleEscapeKeyPress closes tooltip 1`] = `
<body>
  
  
  <span
    class="tooltip"
    data-tooltip="tooltip-id-1"
  >
    
    
    <button
      aria-describedby="tooltip-id-1"
      class="tooltip-trigger"
      data-target="tooltip-id-1"
    >
      Tooltip Button
    </button>
    
    
    <div
      class="tooltip-box"
      data-visible="false"
      id="tooltip-id-1"
      role="tooltip"
      style="left: 0px;"
    >
      
      This is a tooltip.
    
    </div>
    
  
  </span>
  
  
  <span
    class="tooltip"
    data-tooltip="tooltip-id-2"
  >
    
    
    <button
      aria-describedby="tooltip-id-2"
      class="tooltip-trigger"
      data-target="tooltip-id-2"
    >
      Tooltip Button
    </button>
    
    
    <div
      class="tooltip-box"
      id="tooltip-id-2"
      role="tooltip"
    >
      
      This is a tooltip.
    
    </div>
    
  
  </span>
  

</body>
`;

exports[`Tooltip #handleEvent opens tooltip on focus 1`] = `
<body>
  
  
  <span
    class="tooltip"
    data-tooltip="tooltip-id-1"
  >
    
    
    <button
      aria-describedby="tooltip-id-1"
      class="tooltip-trigger"
      data-target="tooltip-id-1"
    >
      Tooltip Button
    </button>
    
    
    <div
      class="tooltip-box"
      data-visible="true"
      id="tooltip-id-1"
      role="tooltip"
      style="left: 0px;"
    >
      
      This is a tooltip.
    
    </div>
    
  
  </span>
  
  
  <span
    class="tooltip"
    data-tooltip="tooltip-id-2"
  >
    
    
    <button
      aria-describedby="tooltip-id-2"
      class="tooltip-trigger"
      data-target="tooltip-id-2"
    >
      Tooltip Button
    </button>
    
    
    <div
      class="tooltip-box"
      id="tooltip-id-2"
      role="tooltip"
    >
      
      This is a tooltip.
    
    </div>
    
  
  </span>
  

</body>
`;

exports[`Tooltip #handleEvent opens tooltip on mouseover 1`] = `
<body>
  
  
  <span
    class="tooltip"
    data-tooltip="tooltip-id-1"
  >
    
    
    <button
      aria-describedby="tooltip-id-1"
      class="tooltip-trigger"
      data-target="tooltip-id-1"
    >
      Tooltip Button
    </button>
    
    
    <div
      class="tooltip-box"
      data-visible="true"
      id="tooltip-id-1"
      role="tooltip"
      style="left: 0px;"
    >
      
      This is a tooltip.
    
    </div>
    
  
  </span>
  
  
  <span
    class="tooltip"
    data-tooltip="tooltip-id-2"
  >
    
    
    <button
      aria-describedby="tooltip-id-2"
      class="tooltip-trigger"
      data-target="tooltip-id-2"
    >
      Tooltip Button
    </button>
    
    
    <div
      class="tooltip-box"
      id="tooltip-id-2"
      role="tooltip"
    >
      
      This is a tooltip.
    
    </div>
    
  
  </span>
  

</body>
`;

exports[`Tooltip API start sets attributes 1`] = `
<body>
  
  
  <span
    class="tooltip"
    data-tooltip="tooltip-id-1"
  >
    
    
    <button
      aria-describedby="tooltip-id-1"
      class="tooltip-trigger"
      data-target="tooltip-id-1"
    >
      Tooltip Button
    </button>
    
    
    <div
      class="tooltip-box"
      id="tooltip-id-1"
      role="tooltip"
    >
      
      This is a tooltip.
    
    </div>
    
  
  </span>
  
  
  <span
    class="tooltip"
    data-tooltip="tooltip-id-2"
  >
    
    
    <button
      aria-describedby="tooltip-id-2"
      class="tooltip-trigger"
      data-target="tooltip-id-2"
    >
      Tooltip Button
    </button>
    
    
    <div
      class="tooltip-box"
      id="tooltip-id-2"
      role="tooltip"
    >
      
      This is a tooltip.
    
    </div>
    
  
  </span>
  

</body>
`;

exports[`Tooltip API stop + Focus/Mouseover does not open tooltip on focus 1`] = `
<body>
  
  
  <span
    class="tooltip"
    data-tooltip="tooltip-id-1"
  >
    
    
    <button
      aria-describedby="tooltip-id-1"
      class="tooltip-trigger"
      data-target="tooltip-id-1"
    >
      Tooltip Button
    </button>
    
    
    <div
      class="tooltip-box"
      id="tooltip-id-1"
      role="tooltip"
    >
      
      This is a tooltip.
    
    </div>
    
  
  </span>
  
  
  <span
    class="tooltip"
    data-tooltip="tooltip-id-2"
  >
    
    
    <button
      aria-describedby="tooltip-id-2"
      class="tooltip-trigger"
      data-target="tooltip-id-2"
    >
      Tooltip Button
    </button>
    
    
    <div
      class="tooltip-box"
      id="tooltip-id-2"
      role="tooltip"
    >
      
      This is a tooltip.
    
    </div>
    
  
  </span>
  

</body>
`;

exports[`Tooltip API stop + Focus/Mouseover does not open tooltip on mouseover 1`] = `
<body>
  
  
  <span
    class="tooltip"
    data-tooltip="tooltip-id-1"
  >
    
    
    <button
      aria-describedby="tooltip-id-1"
      class="tooltip-trigger"
      data-target="tooltip-id-1"
    >
      Tooltip Button
    </button>
    
    
    <div
      class="tooltip-box"
      id="tooltip-id-1"
      role="tooltip"
    >
      
      This is a tooltip.
    
    </div>
    
  
  </span>
  
  
  <span
    class="tooltip"
    data-tooltip="tooltip-id-2"
  >
    
    
    <button
      aria-describedby="tooltip-id-2"
      class="tooltip-trigger"
      data-target="tooltip-id-2"
    >
      Tooltip Button
    </button>
    
    
    <div
      class="tooltip-box"
      id="tooltip-id-2"
      role="tooltip"
    >
      
      This is a tooltip.
    
    </div>
    
  
  </span>
  

</body>
`;

exports[`Tooltip Drop Direction renders bottom 1`] = `
<body>
  
  
  <span
    class="tooltip"
    data-tooltip="tooltip-id-1"
  >
    
    
    <button
      aria-describedby="tooltip-id-1"
      class="tooltip-trigger"
      data-target="tooltip-id-1"
    >
      Tooltip Button
    </button>
    
    
    <div
      class="tooltip-box is-drop-block-end"
      data-visible="true"
      id="tooltip-id-1"
      role="tooltip"
      style="left: 0px;"
    >
      
      This is a tooltip.
    
    </div>
    
  
  </span>
  
  
  <span
    class="tooltip"
    data-tooltip="tooltip-id-2"
  >
    
    
    <button
      aria-describedby="tooltip-id-2"
      class="tooltip-trigger"
      data-target="tooltip-id-2"
    >
      Tooltip Button
    </button>
    
    
    <div
      class="tooltip-box"
      id="tooltip-id-2"
      role="tooltip"
    >
      
      This is a tooltip.
    
    </div>
    
  
  </span>
  

</body>
`;

exports[`Tooltip Drop Direction renders left 1`] = `
<body>
  
  
  <span
    class="tooltip"
    data-tooltip="tooltip-id-1"
  >
    
    
    <button
      aria-describedby="tooltip-id-1"
      class="tooltip-trigger"
      data-target="tooltip-id-1"
    >
      Tooltip Button
    </button>
    
    
    <div
      class="tooltip-box is-drop-inline-start"
      data-visible="true"
      id="tooltip-id-1"
      role="tooltip"
      style="top: 0px;"
    >
      
      This is a tooltip.
    
    </div>
    
  
  </span>
  
  
  <span
    class="tooltip"
    data-tooltip="tooltip-id-2"
  >
    
    
    <button
      aria-describedby="tooltip-id-2"
      class="tooltip-trigger"
      data-target="tooltip-id-2"
    >
      Tooltip Button
    </button>
    
    
    <div
      class="tooltip-box"
      id="tooltip-id-2"
      role="tooltip"
    >
      
      This is a tooltip.
    
    </div>
    
  
  </span>
  

</body>
`;

exports[`Tooltip Drop Direction renders right 1`] = `
<body>
  
  
  <span
    class="tooltip"
    data-tooltip="tooltip-id-1"
  >
    
    
    <button
      aria-describedby="tooltip-id-1"
      class="tooltip-trigger"
      data-target="tooltip-id-1"
    >
      Tooltip Button
    </button>
    
    
    <div
      class="tooltip-box is-drop-inline-end"
      data-visible="true"
      id="tooltip-id-1"
      role="tooltip"
      style="top: 0px;"
    >
      
      This is a tooltip.
    
    </div>
    
  
  </span>
  
  
  <span
    class="tooltip"
    data-tooltip="tooltip-id-2"
  >
    
    
    <button
      aria-describedby="tooltip-id-2"
      class="tooltip-trigger"
      data-target="tooltip-id-2"
    >
      Tooltip Button
    </button>
    
    
    <div
      class="tooltip-box"
      id="tooltip-id-2"
      role="tooltip"
    >
      
      This is a tooltip.
    
    </div>
    
  
  </span>
  

</body>
`;
